packages/puppeteer-core/src/common/TouchPointIdRepository.ts

```diff
@@ -0,0 +1,40 @@
+/**
+ * @license
+ * Copyright 2020 Google Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+interface IdRecord {
+  id: number;
+  inUse: boolean;
+}
+export class TouchPointIdRepository {
```

Comment:
This is to adhere to the way pointer ids are usually chosen: they are re-used. So the first gets id 0, the second gets id 1, and then the first ends, and then the third gets id 0, because 0 is available. See [this](https://w3c.github.io/webdriver/#dfn-get-a-pointer-id) also. Unfortunately I don't have a test that proves that this is necessary, but I believe it is.

Reply:
Let me know :) Would be nice if we could be sure it wasn't necessary

Reply:
So do we keep this?

Reply:
Let's simplify this with the `createIncrementalIdGenerator` (and extract this in utils). 
The spec is a little bit ambiguous of what exactly it is. 
https://developer.mozilla.org/en-US/docs/Web/API/Touch/identifier - This looks like the only API that let's you distinguish between the touches and they are unique per start/mode/end/cancel

Reply:
on it :)